packages/relay/src/lib/services/transactionPoolService/transactionPoolService.ts

```diff
@@ -0,0 +1,132 @@
+// SPDX-License-Identifier: Apache-2.0
+
+import { Transaction } from 'ethers';
+import { Logger } from 'pino';
+
+import { IExecuteTransactionEventPayload } from '../../types/events';
+import {
+  PendingTransactionStorage,
+  TransactionPoolService as ITransactionPoolService,
+} from '../../types/transactionPool';
+
+/**
+ * Service implementation that orchestrates pending transaction management.
+ * Acts as a facade for the underlying storage layer and coordinates transaction lifecycle.
+ */
+export class TransactionPoolService implements ITransactionPoolService {
+  /**
+   * The logger used for logging transaction pool operations.
+   *
+   * @private
+   */
+  private readonly logger: Logger;
+
+  /**
+   * The storage implementation for managing pending transactions.
+   *
+   * @private
+   */
+  private readonly storage: PendingTransactionStorage;
+
+  /**
+   * Creates a new TransactionPoolService instance.
+   *
+   * @param storage - The storage backend for pending transactions.
+   * @param logger - The logger instance for transaction pool operations.
+   */
+  constructor(storage: PendingTransactionStorage, logger: Logger) {
+    this.storage = storage;
+    this.logger = logger.child({ name: 'transaction-pool-service' });
+  }
+
+  /**
+   * Saves a transaction into the transaction pool for the given address.
+   *
+   * @param address - The account address that submits the transaction.
+   * @param tx - The transaction object to be stored.
+   * @returns A promise that resolves once the transaction is stored.
+   */
+  async saveTransaction(address: string, tx: Transaction): Promise<void> {
+    const txHash = tx.hash;
+
+    if (!txHash) {
+      throw new Error('Transaction hash is required for storage');
```

Comment:
q: here we're using a **signed** ethers Transaction, which always should have a hash, right?